### ISSUE TITLE
Unable to install

### ISSUE THREAD

**Issue by nicholst** [MRtrix3/mrpeek#12]
Really wanting to give this a try but failing at step 2.

1. install MRtrix3 from source (did it via MacOS application package installer, no problem)
2. clone this repo and set it up as a module (cloned, but build/configure dox aren't working)

The documentation on modules seems to be out of date.  While the page - https://mrtrix.readthedocs.io/en/latest/tips_and_tricks/external_modules.html - is clear and detailed, it depends on a `mrtrix3/build` program that isn't in the install.  My `/usr/local/mrtrix3` directory has just 4 folders (bin, lib, matlab, share) and a file with symlinks, but no build.  In fact,  `find /usr/local/mrtrix3 -name '*build*'` finds no such program anywhere.

Other documentation https://www.mrtrix.org/developer-documentation/build_page.html points to a `configure` script, but likewise `find /usr/local/mrtrix3 -name '*configure*'` finds nothing.

Any hints?

**Comment by jdtournier** [comment]
OK, the issue is that you've not installed MRtrix3 from source. The macOS installer fetches precompiled binaries. Try this:
```
git clone https://github.com/MRtrix3/mrtrix3.git
cd mrtrix3
./configure -noshared -nogui
git clone https://github.com/MRtrix3/mrpeek.git
cd mrpeek
../build
```
That should produce the executable in `bin/mrpeek`, without needing the MRtrix3 shared library.

**Comment by jdtournier** [comment]
If you're missing the `eigen3` dependency, you can grab it through `git `too - it's a header-only library. Just modify the `configure` call like this:
```
git clone https://gitlab.com/libeigen/eigen.git
cd eigen; 
git checkout 3.3.7; 
cd ..
CFLAGS="-idirafter $(pwd)/eigen" ./configure -noshared -nogui
```
and proceed with the next line.

[EDIT updated to grab specific 3.3.7 tag, since things don't seem to work with `eigen`'s `master` branch right now]

**Comment by nicholst** [comment]
Thank you!  This was enough for me to get this running... super!  Still getting used to iTerm2 but I'm sure I'll manage.

There's still just one typo on the instructions.  You have
```
git clone https://github.com/MRtrix3/mrpeek.git
cd mrpeek
../build
```
when you mean `../mritrx3/build` right?

**Comment by jdtournier** [comment]
Good to hear it!

Regarding the 'typo': what you did is correct if you install the main `mrtrix3` and module `mrpeek` repos side by side in the same folder, as is actually [recommended on our own docs](https://mrtrix.readthedocs.io/en/latest/tips_and_tricks/external_modules.html). The instructions I provided on that tweet were for a quick & dirty build of `mrpeek` only, and implied the `mrpeek` repo would be cloned _inside_ the main `mrtrix3` one - in which case the full path to the main `build` script from the `mrpeek` repo folder would be `../build`. There's no hard requirement about the locations of the modules relative to the main repo - we just need to invoke the main repo's `build` script (wherever that may be) from within the module's folder.

We've actually added these instructions on the [README](https://github.com/MRtrix3/mrpeek/blob/master/README.md) for this repo - but maybe we should  amend them to match the docs more closely and avoid any confusion of this nature...
